### PR TITLE
improve: consolidate query logic

### PR DESCRIPTION
**Pull request by james-a-morris**
This PR leverages OOP to consolidate the query logic of each Blockchain Query class. (Arbitrum, Ethereum, Optimism, Boba, Polygon)

### CODE REVIEW

File: src/relayFeeCalculator/chain-queries/boba.ts [hunk 1]
```diff
@@ -1,53 +1,29 @@
...
+    super(
+      provider,
+      symbolMapping,
+      spokePoolAddress,
+      usdcAddress,
+      simulatedRelayerAddress,
+      gasMarkup,
+      logger,
+      coingeckoProApiKey,
+      utils.parseUnits("1", 9)
```
Comment (nicholaspai) [inline src/relayFeeCalculator/chain-queries/boba.ts:26]: Why does only boba parse units here?

Comment (nicholaspai) [reply src/relayFeeCalculator/chain-queries/boba.ts:26]: I know this isn't your code change but perhaps you understand deeply

**Review by nicholaspai — APPROVED** [review]
This is awesome!!

**Review by dohaki — APPROVED** [review]
👏

**Pull request merged**